FilePersistence/src/version_control/Merge.cpp

```diff
@@ -43,6 +43,11 @@ namespace FilePersistence {
 
 bool Merge::commit(const Signature& author, const Signature& committer, const QString& message)
 {
+	if (stage_ == Stage::Committed)
```

Comment:
Hmm, this is not the right place for this check. You need to do this in the place where `commit()` is called and not call it in case `AlreadyUpToDate` is the case. Please reset this commit and make a new one (instead of just fixing this code here, by removing it). So the PR should still have one commit afterwards.